static/js/marathon.js

```diff
@@ -152,25 +161,21 @@ let app = new Vue({
                     this.visitedCheckpoints.push(page);
                 }
             }
-                       
-            if (this.clicksRemaining === 0) {
-                await this.finish(1);
-            }
-
-            setMargin();
-            
             
             if (hitcheckpoint) {
                 let el = this.checkpoints.shift()
                 console.log(el)
                 this.activeCheckpoints[checkpointindex] = el
 
                 conf();
-            }
 
-            if (!this.reachedstop && this.checkpointMarkReached) {
-                this.showStop = true
-                this.reachedstop = !this.reachedstop
+                if (!this.reachedstop && this.checkpointMarkReached) {
+                    this.showStop = true
+                    this.reachedstop = true
+                }
+
+            } else if (this.clicksRemaining === 0) {
+                this.finish(1);
```

Comment:
Was there a specific reason to make pageCallback not async?

Reply:
Otherwise, agreed with moving finish to last, it makes more sense

Reply:
The only async logic inside pageCallback is that for the `saveRun` API call inside the `finish` method. Since this is a terminal non-blocking call it doesn't make sense to make pageCallback async otherwise we would need to `await` it everywhere we want to call it synchronously (which is important when rendering components that depend on the completion of the Vue tick cycle e.g. time-box margin). Moreover, since `await` can only be used inside async functions, we would be forced to make the calling function async as well then.